### ISSUE TITLE
Resolve data for multiple named views

### ISSUE THREAD

**Issue by jkossis** [angular-ui/ui-router#2302]
Is there any way to resolve data or inject data into each view under a state's views? I only see controller and templateUrl. Below is an example:

```
$stateProvider
  .state('report',{
    views: {
      'filters': {
        templateUrl: 'report-filters.html',
        controller: function($scope){ ... controller stuff just for filters view ... }
      },
      'tabledata': {
        templateUrl: 'report-table.html',
        controller: function($scope){ ... controller stuff just for tabledata view ... }
      },
      'graph': {
        templateUrl: 'report-graph.html',
        controller: function($scope){ ... controller stuff just for graph view ... }
      }
    }
  })

```

**Comment by eddiemonge** [comment]
Resolve will work for all views so you only need the one.

**Comment by jkossis** [comment]
I understand that, I guess let me rephrase. I want to pass a json object
into each controller of each view under views. If I were to use the parent
state's resolve, I would need to know the names of what i want to resolve
so I can inject them in the corresponding controllers...I do not know the
names or amount of things I want to resolve ahead of time, it is dynamic.
Does that make sense?

Kind Regards,
Jason

**Comment by eddiemonge** [comment]
How are you going to inject them into the controller if you don't know the names?

**Comment by jkossis** [comment]
Well is there a way to inject it as a param? I could give it a generic
name, not hung up on that, but how can I pass it from the state to one of
the nested views? For example, I could have three <div ui-view="test">,
giving myself three templates of "test" in the state. But I want to be able
to pass each a config object that could be different.

Kind Regards,
Jason

**Comment by aL3xa** [comment]
I can get access via `$resolve`
